app/models/race.rb

```diff
@@ -47,6 +47,10 @@ def paid_participants_count
     participants.with_payment.size
   end
 
+  def paid_not_cancelled_count
+    participants.with_payment.not_archived.not_cancelled.size
+  end
+
```

Comment:
This does not feel like a method that should be on the Race model. Better for. decorator.